File: 프로그래머스/unrated/181880. 1로 만들기/1로 만들기.java
//1. 일반 풀이
class Solution {
    public int solution(int[] num_list) {
        int answer = 0;
        for(int num : num_list){
            while(num != 1){
             num = num % 2 == 0 ? num / 2 : (num-1)/2;
              answer++;
            }
        }
        return answer;
    }
}

//2. 스트림 풀이 //굳이
// import java.util.Arrays;
// class Solution {
//     public int solution(int[] num_list) {
//         return Arrays.stream(num_list).map(num -> {
//             int cnt = 0; 
//             while(num != 1) {
//                 num = num % 2 == 0 ? num / 2 : (num-1)/2;
//                 cnt++;
//             }
//             return cnt;
//         }).sum();
//     }
// }

File: 프로그래머스/0/181880. 1로 만들기/1로 만들기.cpp
#include <vector>

using namespace std;

void dfs(int n, int& answer){
    if(n == 1){
        return;
    }
    dfs(n / 2 , ++answer);
}
int solution(vector<int> num_list) {
    int answer = 0;
    for(int i : num_list){
        dfs(i, answer);
    }
    return answer;
}